scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/ProjectManagerImpl.java

```diff
@@ -0,0 +1,68 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.persistence.NonUniqueResultException;
+import javax.transaction.Transactional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.util.NestedServletException;
+
+import edu.asu.diging.scriptoocloud.core.data.ProjectRepository;
+import edu.asu.diging.scriptoocloud.core.model.Project;
+import edu.asu.diging.scriptoocloud.core.model.impl.ProjectImpl;
+import edu.asu.diging.scriptoocloud.core.service.ProjectManager;
+
+@Transactional
+@Service
+public class ProjectManagerImpl implements ProjectManager {
+	
+	private static final Logger LOGGER=LoggerFactory.getLogger(ProjectManager.class);
+
+    @Autowired
+    private ProjectRepository projectRepo;
+    
+    /* (non-Javadoc)
+     * @see edu.asu.diging.scriptoocloud.core.service.impl.ProjectManager#createProject(java.lang.String, java.lang.String)
+     */
+    @Override
+    public Project createProject(String name, String description) {
+        Project project = new ProjectImpl();
+        project.setName(name);
+        project.setDescription(description);
+        project.setCreationDate(ZonedDateTime.now());
+        
+        return projectRepo.save((ProjectImpl)project);
+    }
+    
+    /* (non-Javadoc)
+     * @see edu.asu.diging.scriptoocloud.core.service.impl.ProjectManager#deleteProject(java.lang.String)
+     */
+    @Override
+    public void deleteProject(long id) {
+   
+        try {
+        	Project project = projectRepo.findById(id);
+        	projectRepo.delete((ProjectImpl)project);
+        } catch (IllegalArgumentException e) {
```

Comment:
couple of things:
- this is only thrown when you pass null I believe, which is something you can check for and shouldn't have to catch
- if you do have to catch an exception, you have to log it. Right now you are swallowing the exception.

Reply:
This was in place when I removed by search. Since it is a button, and always takes a value (never null), is it appropriate to remove the try catch, and not check for null?

If I still have to check for null, would I say "if null, give user an error" in the form of a page? (as opposed to logging).

Reply:
Just because you are using this method through a button call on the page, doesn't mean everybody will use the method like this ;) meaning, you can't expect to never be passed null. I would check and then raise an exception if id is null. In your controller, I would catch the exception and then show an error message that something went wrong.

Reply:
Ok. So I think I would do something similar to this stack solution https://stackoverflow.com/questions/31334326/handling-null-objects-and-throwing-exceptions-in-streams , where I would be creating my own exception. 

Then, in my created exception, show the error message on the page. My psuedo code without creating the exception class looks like so at this moment:

`public void deleteProject(long id) {
   
        try {
        	// long is a primitive data type, so we must convert to an object data type Long to check for null. 
        	Long checkId = id;
        	if(checkId == null) {
        		throw new myException();
        	}else {
        		projectRepo.deleteById(id);
        	}
        } catch (myException exc) {
        	throw myCheckedException();
        	logger.error("Project does not exist."); 
        }
    }`
    
    Is this what you are suggesting? I can go through with creation and do a new pr if so.

Reply:
ya, something like that. `IllegalArgumentException` is, however, a runtime exception if I'm not mistaken. And since it's only thrown when you pass null (I believe), you wouldn't even have to catch it if you check for null first.